### PR TITLE
modify simpleprometheus receiver to use httpclientsetting

### LINKED ISSUE #9571: Simple Prometheus receiver should use HTTP Client Settings
As seen on #9570, the simple Prometheus receiver should make use of the standard HTTP Client Settings, so that it can take advantage of the standard features that it provides, such as access to the client authenticators.

### PR DESCRIPTION
**Pull request by fatsheep9146**
Signed-off-by: Ziqi Zhao <zhaoziqi9146@gmail.com>

**Description:** <Describe what has changed.>
Fix #9571

**Link to tracking Issue:** <Issue number if applicable>
#9571

### CODE REVIEW

**Comment by fatsheep9146** [comment]
ping @codeboten @dmitryax , I already modify the simpleprometheusreceiver to use confighttp.HttpClientSetting, but there is still one problem that I'm not sure, since the origin tls config of simpleprometheusreceiver has totally different key name from the tls config of confighttp.HttpClientSetting. 

for example, the tls config of simpleprometheusreceiver is 
```
        endpoint: "172.17.0.5:9153"
        tls_enabled: true
        tls_config:
            ca_file: "/path/to/ca"
            cert_file: "/path/to/cert"
            key_file: "/path/to/key"
            insecure_skip_verify: true
```

the tls config of confighttp.HttpClientSetting is 
```
    endpoint: myserver.local:55690
    tls:
      insecure: false
      ca_file: server.crt
      cert_file: client.crt
      key_file: client.key
```

What is the best way to release those changes safely? Does this have any guideline like https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#breaking-changes?

**Comment by dmitryax** [comment]
Please keep support of `tls_config` as a deprecated option for now and show a warning log if it's being used. It can be removed later

**Comment by fatsheep9146** [comment]
> Please keep support of `tls_config` as a deprecated option for now and show a warning log if it's being used. It can be removed later
@dmitryax 

I add `tls_config` , `tls_enable` back to configuration, add a new function `getPrometheusConfigWrapper` in the basic of `getPrometheusConfig`. 

If user specify `tls_enable` , I print a warning log and use the info in `tls_config` to create httpClientSetting config then pass to `getPrometheusConfig`. 

In this way, I think we can just delete `tls_enable`, `tls_config` field, function `getPrometheusConfigWrapper`  and usage `getPrometheusConfig` directly.

**Comment by dmitryax** [comment]
Please add a changelog entry

File: receiver/simpleprometheusreceiver/receiver.go [hunk 1]
```diff
@@ -60,8 +61,30 @@ func (prw *prometheusReceiverWrapper) Start(ctx context.Context, host component.
...
+}
+
 func getPrometheusConfig(cfg *Config) (*prometheusreceiver.Config, error) {
-	var bearerToken string
+	var (
+		bearerToken string
+		scheme      = "http"
+		out         = &prometheusreceiver.Config{}
+		httpConfig  = configutil.HTTPClientConfig{}
+	)
```
Comment (dmitryax) [inline receiver/simpleprometheusreceiver/receiver.go:86]: I would avoid using this style inside functions. I doesn't improve readability and moves variable definitions farther away from its usage. Please revert this block

**Review by dmitryax — COMMENTED** [review]
One nit, otherwise LGTM

**Comment by fatsheep9146** [comment]
ping @codeboten @dmitryax I think pr is ready to merge :-)

File: receiver/simpleprometheusreceiver/receiver.go [hunk 2]
```diff
@@ -78,13 +95,17 @@ func getPrometheusConfig(cfg *Config) (*prometheusreceiver.Config, error) {
 
 	scheme := "http"
 
-	if cfg.TLSEnabled {
+	tlsConfig, err := cfg.TLSSetting.LoadTLSConfig()
```
Comment (dmitryax) [inline receiver/simpleprometheusreceiver/receiver.go:98]: Shouldn't `cfg.HTTPClientSettings.TLSSetting.Insecure` be set to `true` by default to get `tlsConfig = nil`?

**Review by dmitryax — COMMENTED** [review]
Did you test the receiver? I doesn't look like working for regular http endpoints

File: receiver/simpleprometheusreceiver/receiver.go (hunk 2 above)
Comment (fatsheep9146) [reply receiver/simpleprometheusreceiver/receiver.go:99]: > Did you test the receiver? I doesn't look like working for regular http endpoints

Sorry, I didn't test it sufficiently, so this time I add another test Function `TestGetPrometheusConfigWrapper` to test the new function I add getPrometheusConfigWrapper, in 4 cases

- `tls_enable` configured true, `tls` default config,  which results to https
- `tls_enable` configured false(default), `tls` default config,  which results to http
- `tls_enable` configured false(default), `tls` config to use client security,  which results to https
- `tls_enable` configured false(default), `tls` config to use client security, and specify ca path,  which results to https

**Review by dmitryax — APPROVED** [review]

**Pull request merged**